app/src/main/java/com/fstyle/structure_android/utils/Validator.java

```diff
@@ -0,0 +1,359 @@
+        }
+        return mContext.getString(mAllErrorMessage.keyAt(index));
+    }
+
+    public void initNGWordPattern() {
+        Observable.create(new Observable.OnSubscribe<Pattern>() {
+            @Override
+            public void call(Subscriber<? super Pattern> subscriber) {
+                try {
+                    BufferedReader reader = new BufferedReader(
+                            new InputStreamReader(mContext.getAssets().open("ng-word")));
+                    StringBuffer buffer = new StringBuffer();
+                    String line;
+                    while ((line = reader.readLine()) != null) {
+                        if (buffer.length() == 0) {
+                            buffer.append("(").append(line.toLowerCase(Locale.ENGLISH));
+                        } else {
+                            buffer.append("|").append(line.toLowerCase(Locale.ENGLISH));
+                        }
+                    }
+                    subscriber.onNext(Pattern.compile(buffer.append(")").toString()));
+                } catch (IOException e) {
+                    subscriber.onError(e);
+                }
+            }
+        })
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(new Action1<Pattern>() {
+                    @Override
+                    public void call(Pattern pattern) {
+                        mNGWordPattern = pattern;
+                    }
+                }, new Action1<Throwable>() {
+                    @Override
+                    public void call(Throwable throwable) {
+                        throw new ValidationException("Error when open ng-word", throwable);
+                    }
+                });
+    }
+
+    @ValidMethod(type = { ValidType.NG_WORD })
+    public String validateNGWord(String str) {
+        if (mNGWordPattern == null) {
+            throw new ValidationException(
+                    "NGWordPattern is null!!! \n Call initNGWordPattern() before call this method!",
+                    new NullPointerException());
+        }
+        boolean isValid =
+                TextUtils.isEmpty(str) || !mNGWordPattern.matcher(str.toLowerCase(Locale.ENGLISH))
+                        .find();
+        return isValid ? "" : mContext.getString(mAllErrorMessage.valueAt(ValidType.NG_WORD));
+    }
+
+    @ValidMethod(type = { ValidType.VALUE_RANGE_0_100 })
+    public String validateValueRangeFrom0to100(int value) {
+        boolean isValid = value >= 0 && value <= 100;
+        return isValid ? ""
+                : mContext.getString(mAllErrorMessage.valueAt(ValidType.VALUE_RANGE_0_100));
+    }
+
+    @Documented
+    @Retention(RetentionPolicy.RUNTIME)
+    @Target({ ElementType.FIELD, ElementType.METHOD })
+    public @interface Optional {
+
+    }
+
+    @IntDef({ ValidType.NG_WORD, ValidType.VALUE_RANGE_0_100 })
+    public @interface ValidType {
+        int NG_WORD = 0x00;
```

Comment:
ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.